packages/webapp/src/hooks/useOfflineCheck.ts

```diff
@@ -0,0 +1,73 @@
+import { useState, useEffect } from "react";
+import { toast } from "sonner";
+
+async function performNetworkCheck() {
+  const endpoint = "https://www.google.com";
+  try {
+    const response = await fetch(endpoint, {
+      method: "HEAD",
+      cache: "no-cache",
+      mode: "no-cors",
+    });
+    return response.ok;
+  } catch (error) {
+    throw error; // Rethrow for retries
```

Comment:
This is redundant, you could remove the whole `try-catch` statement and it would work the same.